skiko/src/awtMain/objectiveC/macos/Drawlayer.mm

```diff
@@ -362,7 +362,14 @@ JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_HardwareLayer_getWindowHandle(J
 {
     @autoreleasepool {
         LayerHandler *layer = findByObject(env, component);
-        return (jlong) (__bridge void*) layer.window;
+
+        NSWindow *window = layer.window;
+
+        if (window) {
+            return (jlong) (__bridge_retained void*) window;
+        } else {
+            return 0;
```

Comment:
In which cases it might be 0?

Reply:
Assigning a weak property to a strong ref will unwrap it and writing `nil` inside, if the object was deallocated.